lib/matplotlib/axis.py

```diff
@@ -2029,6 +2029,9 @@ def set_ticks(self, ticks, labels=None, *, minor=False, **kwargs):
         other limits, you should set the limits explicitly after setting the
         ticks.
         """
+        if labels is None and kwargs:
+            raise ValueError('labels keyword argument cannot be None when '
```

Comment:
Minor nit: labels is not (necessarily) a keyword argument, it is in fact positional or keyword. I would recommend just taking out that distinction.